packages/web-components/package.json

```diff
@@ -0,0 +1,100 @@
+{
+  "name": "@microsoft/fast-components-msft",
+  "description": "A library of Fluent Web Components",
+  "sideEffects": false,
+  "version": "1.6.0",
+  "author": {
+    "name": "Microsoft",
+    "url": "https://discord.gg/FcSNfg4"
+  },
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Microsoft/fast.git"
+  },
+  "bugs": {
+    "url": "https://github.com/Microsoft/fast/issues/new/choose"
+  },
+  "main": "dist/esm/index.js",
+  "types": "dist/fast-components-msft.d.ts",
+  "unpkg": "dist/fast-components-msft.min.js",
+  "scripts": {
+    "clean:dist": "node ./build/clean.js dist",
+    "doc": "api-extractor run --local",
+    "doc:ci": "api-extractor run --local",
+    "build": "tsc -p ./tsconfig.json && rollup -c && npm run doc",
+    "dev": "tsc -p ./tsconfig.json -w",
+    "tdd": "npm run dev & npm run test-chrome:watch",
+    "prepare": "yarn clean:dist && yarn build",
+    "prettier": "prettier --config ../../prettier.config.js --write \"**/*.{ts,html}\"",
+    "prettier:diff": "prettier --config ../../prettier.config.js \"**/*.{ts,html}\" --list-different",
+    "eslint": "eslint . --ext .ts",
+    "eslint:fix": "eslint . --ext .ts --fix",
+    "start": "start-storybook -p 6006",
+    "build-storybook": "build-storybook",
+    "test": "yarn test-chrome:verbose && yarn doc:ci",
+    "test-node": "mocha --reporter min --exit dist/esm/__test__/setup-node.js './dist/esm/**/*.spec.js'",
+    "test-node:verbose": "mocha --reporter spec --exit dist/esm/__test__/setup-node.js './dist/esm/**/*.spec.js'",
+    "test-chrome": "karma start karma.conf.js --browsers=ChromeHeadlessOpt --single-run --coverage",
+    "test-chrome:verbose": "karma start karma.conf.js --browsers=ChromeHeadlessOpt --single-run --coverage --reporter=mocha",
+    "test-chrome:watch": "karma start karma.conf.js --browsers=ChromeHeadlessOpt --coverage --watch-extensions js",
+    "test-chrome:debugger": "karma start karma.conf.js --browsers=ChromeDebugging",
+    "test-chrome:verbose:watch": "karma start karma.conf.js --browsers=ChromeHeadlessOpt --coverage --watch-extensions js --reporter=mocha ",
+    "test-chrome:verbose:debugger": "karma start karma.conf.js --browsers=ChromeDebugging --reporter=mocha",
+    "test-firefox": "karma start karma.conf.js --browsers=FirefoxHeadless --single-run --coverage",
+    "test-firefox:verbose": "karma start karma.conf.js --browsers=FirefoxHeadless --single-run --coverage --reporter=mocha"
+  },
+  "devDependencies": {
+    "@microsoft/api-extractor": "^7.7.13",
```

Comment:
Having different versions of build and types dependencies between this package and the rest of the repo is going to cause some headaches:
- We use [syncpack](https://www.npmjs.com/package/syncpack) to validate that all packages have matching dependency versions (otherwise we're relying on people noticing duplicates in yarn.lock, and things regularly used to sneak through that way and cause problems later). Unfortunately, syncpack currently has no option to ignore a particular package or allow multiple versions for certain deps, so we'll need to either find a different tool, or add this feature to syncpack (original or fork).
- Largely due to the common practice of type packages using `*` as the version spec when depending on other type packages, if multiple versions of the same types are installed, it can be unpredictable which one will be picked up
- I think having multiple versions of typescript or other dependencies around can also cause problems (don't remember specifics as I haven't had to deal with this for awhile)

In some cases, we may be able to resolve the dependency mismatches without too much trouble, but in other cases it will require work in either web-components or the entire repo to handle a nontrivial upgrade or downgrade.

Reply:
I'll try to reconcile these, but one we won't be able to reconcile is Typescript. There are ways to prevent hoisting of certain packages to the root which in the case of Typescript may be beneficial for the short term.

Reply:
*most* of these have been reconciled. I did add a flag to prevent hoisting of the TS related dependencies here, so that should leave everything else undisturbed and install those locally for use in this package: https://classic.yarnpkg.com/blog/2018/02/15/nohoist/